recipes/automake/all/conanfile.py

```diff
@@ -1,140 +1,181 @@
-
-    _autotools = None
-
     @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+    def _settings_build(self):
+        # TODO: Remove for Conan v2
+        return getattr(self, "settings_build", self.settings)
 
     @property
-    def _version_major_minor(self):
-        [major, minor, _] = self.version.split(".", 2)
-        return '{}.{}'.format(major, minor)
+    def win_bash(self):
+        return self._settings_build.os == "Windows"
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def configure(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+        try:
+            del self.settings.compiler.libcxx  # for plain C projects only
+        except ValueError:
+            pass
+        try:
+            del self.settings.compiler.cppstd  # for plain C projects only
+        except ValueError:
+            pass
+
+    def layout(self):
+        basic_layout(self, src_folder="automake")
 
     def requirements(self):
         self.requires("autoconf/2.71")
-        # automake requires perl-Thread-Queue package
-
-    def build_requirements(self):
-        if hasattr(self, "settings_build"):
-            self.build_requires("autoconf/2.71")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.requires("m4/1.4.19")
 
     def package_id(self):
-        del self.info.settings.arch
-        del self.info.settings.compiler
-        del self.info.settings.build_type
+        try:
+            del self.info.settings.arch
+        except ValueError:
+            pass
+        try:
+            del self.info.settings.compiler
+        except ValueError:
+            pass
+        try:
+            del self.info.settings.build_type
+        except ValueError:
+            pass
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def build_requirements(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            self.tool_requires("msys2/cci.latest")
 
-    @property
-    def _datarootdir(self):
-        return os.path.join(self.package_folder, "res")
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.extend([
+            "--datarootdir=${prefix}/res",
+        ])
```

Comment:
```suggestion
        tc.configure_args.append("--datarootdir=${prefix}/res")
```

Reply:
Done this way for easy extendable without minimal code changes. If it requires additional arguments in the future. hence the comma at the end of the line.